packages/proxy/lib/http/request-middleware.ts

```diff
@@ -30,6 +31,31 @@ const ExtractIsAUTFrameHeader: RequestMiddleware = async function () {
   this.next()
 }
 
+const CopyCrossOriginCookies: RequestMiddleware = function () {
+  const currentAUTUrl = this.getAUTUrl()
+
+  if (!this.config.experimentalSessionAndOrigin || !currentAUTUrl) {
+    return this.next()
+  }
+
+  const sameSiteContext = getSameSiteContext(
+    currentAUTUrl,
+    this.req.proxiedUrl,
+    this.req.isAUTFrame,
+  )
+
+  const cookies = this.getCookieJar().getCookies(this.req.proxiedUrl, sameSiteContext)
+  const existingCookies = this.req.headers['cookie'] ? [this.req.headers['cookie']] : []
+  const cookiesToAdd = cookies.map((cookie) => `${cookie.key}=${cookie.value}`)
+
+  this.debug('existing cookies on request: %s', existingCookies.join('; '))
+  this.debug('add cookies to request: %s', cookiesToAdd.join('; '))
+
+  this.req.headers['cookie'] = existingCookies.concat(cookiesToAdd).join('; ')
```

Comment:
I looked into this and it appears that if there are two or more cookies with the same key, the first cookie found is preferred. I changed it to prepend the added cookies, since I think it's correct that they take preference. Fixed it and added a test in [`235764f` (#22320)](https://github.com/cypress-io/cypress/pull/22320/commits/235764f07d03324fead704de8bfe0a229a70d9b6).